xbmc/playlists/PlayListXSPF.cpp

```diff
@@ -0,0 +1,121 @@
+/*
+ *  Copyright (C) 2018 Tyler Szabo
+ *  Copyright (C) 2018 Team Kodi
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "PlayListXSPF.h"
+
+#include "utils/log.h"
+#include "utils/URIUtils.h"
+#include "utils/XBMCTinyXML.h"
+#include "URL.h"
+
+using namespace PLAYLIST;
+
+static constexpr char const* LOCATION_TAGNAME = "location";
+static constexpr char const* PLAYLIST_TAGNAME = "playlist";
+static constexpr char const* TITLE_TAGNAME = "title";
+static constexpr char const* TRACK_TAGNAME = "track";
+static constexpr char const* TRACKLIST_TAGNAME = "trackList";
+
+CPlayListXSPF::CPlayListXSPF(void) = default;
+
+CPlayListXSPF::~CPlayListXSPF(void) = default;
+
+static std::string GetXMLText(const TiXmlElement* const pXmlElement)
```

Comment:
> Also, wrap this help in an anonymous namepace to avoid symbol collisions

Marking a global function `static` is equivalent to an anonymous namespace

Reply:
This discussion collapsed with my latest revision; but I'd still like to know which pattern to use. It seems according to [this SO post](https://stackoverflow.com/questions/154469/unnamed-anonymous-namespaces-vs-static-functions) that the anonymous namespace might be a good choice; though I'd argue the best choice is the one that'll be most consistent with the direction the codebase is intending to go.

Reply:
We do not have guidelines on this (yet), so you are free to do what you like - as it is a new file you don't have to match up with anything. My personal choice would be anonymous namespaces though, and from the looks of it @garbear seems to agree.

Reply:
I like the anonymous namespace solution; it also seems like it'd be good to put the constants in there, too.